### PR TITLE
DOC: Add doc examples to np.char

### PR DESCRIPTION
**Pull request by navyagarwal**
Issue #21351 

I'm adding doc examples to `np.char` methods, and will make further commits here as I add examples to other methods where they are missing.

(This PR is made as part of OSD @ GHC 2023 ✨)

### CODE REVIEW

**Review by rossbar — APPROVED** [review]
Great to see you here @navyagarwal :tada: 

This LGTM - it might be interesting to replace the last example with something other than a single character, e.g. `np.char.add(b, "python")`. In practice users should never really be calling this function directly, but I don't think there's harm in adding to the examples. Thanks @navyagarwal !

**Review by bsipocz — APPROVED** [review]
Thanks @navyagarwal for your first contribution to Numpy!

**Comment by ngoldbaum** [comment]
> In practice users should never really be calling this function directly

This is actually the only way to add two string arrays in NumPy:

```
In [1]: import numpy as np

In [2]: np.array(["hello", "hello"]) + np.array(["world", "world"])
---------------------------------------------------------------------------
UFuncTypeError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 np.array(["hello", "hello"]) + np.array(["world", "world"])

UFuncTypeError: ufunc 'add' did not contain a loop with signature matching types (dtype('<U5'), dtype('<U5')) -> None

In [3]: np.char.add(np.array(["hello", "hello"]), np.array(["world", "world"]))
Out[3]: array(['helloworld', 'helloworld'], dtype='<U10')
```

When we add a "add" ufunc for unicode strings then what you're saying is true, but string support isn't even at that level yet :upside_down_face:

File: numpy/core/defchararray.py [hunk 1]
```diff
@@ -318,6 +318,21 @@ def add(x1, x2):
         Output array of `bytes_` or `str_`, depending on input types
         of the same shape as `x1` and `x2`.
 
+    Examples
+    --------
+    >>> a = np.array(["a", "b", "c"])
+    >>> np.char.add("0", a)
+    array(['0a', '0b', '0c'], dtype='<U2')
+
```
Comment (mdhaber) [inline numpy/core/defchararray.py:326]:
```suggestion
```
If there are no comments between the examples, probably should either remove the whitespace or add `>>>` to the blank lines so there are not separate code blocks.

File: numpy/core/defchararray.py [hunk 2]
```diff
@@ -318,6 +318,21 @@ def add(x1, x2):
...
+    >>> a = np.array(["a", "b", "c"])
+    >>> b = np.array(["d", "e", "f"])
+    >>> np.char.add(a, b)
+    array(['ad', 'be', 'cf'], dtype='<U2')
+
```
Comment (mdhaber) [inline numpy/core/defchararray.py:331]:
```suggestion
```

File: numpy/core/defchararray.py [hunk 3]
```diff
@@ -1530,6 +1569,22 @@ def split(a, sep=None, maxsplit=None):
     --------
     str.split, rsplit
 
+    Examples
+    --------
+    >>> arr = np.array(['a b c', 'd e f', 'g h'])
+    >>> np.char.split(arr)
+    array([list(['a', 'b', 'c']), list(['d', 'e', 'f']), list(['g', 'h'])],
+      dtype=object)
+
```
Comment (mdhaber) [inline numpy/core/defchararray.py:1578]:
```suggestion
```

File: numpy/core/defchararray.py [hunk 4]
```diff
@@ -1530,6 +1569,22 @@ def split(a, sep=None, maxsplit=None):
...
+    >>> arr = np.array(['r,g,b', 'x;y', 'z'])
+    >>> np.char.split(arr, ",")
+    array([list(['r', 'g', 'b']), list(['x;y']), list(['z'])], dtype=object)
+
```
Comment (mdhaber) [inline numpy/core/defchararray.py:1582]:
```suggestion
```

**Review by mdhaber — APPROVED** [review]
This looks useful to me as-is. Make-better might be to add short notes about what each example shows, but it is not difficult to infer their purpose. I've added a few comments and additional examples I think would be illustrative, but this PR would improve the documentation as-is if there is not interest in spending more time on it.

**Comment by mdhaber** [comment]
Looks like there are merge conflicts, and examples have been added to these functions in the meantime, so I'll go ahead and close. Thanks @navyagarwal.

**Pull request closed without merging**